src/core/thread/mle_router.cpp

```diff
@@ -2853,7 +2855,7 @@ void MleRouter::HandleDiscoveryRequest(const Message &aMessage, const Ip6::Messa
     otError                      error = OT_ERROR_NONE;
     Tlv                          tlv;
     MeshCoP::Tlv                 meshcopTlv;
-    MeshCoP::DiscoveryRequestTlv discoveryRequest;
+    MeshCoP::DiscoveryRequestTlv discoveryRequest{};
```

Comment:
I guess you are using `{ }` to zero-initialze the object.

I would suggest not doing this and instead setting the legth explicitly to zero in the code.

Two reasons:
- It's more explicit and clear in the code (also we just care about length and not rest of field to be zero)
- I don't  know in detail the rules on different initializations in different version of c++, but from what I recall, it's a bit messy/tricky. I think if we  add a default constructor (or just another constructor) to this `Tlv` or any of its sub-classes, the behavior may change and we may not get the zero initialization of object anymore.